docs/ARCHITECTURE.md

```diff
@@ -1,54 +1,114 @@
 
 <div align="center">
 
 ```mermaid
   graph TB
-      c[ecChronos Instance]---C((Cassandra Node));
-
-      a[ecChronos Instance]---A((Cassandra Node));
-
-      A((Cassandra Node))===B((Cassandra Node));
-
-      A((Cassandra Node))===C((Cassandra Node));
-
-      C((Cassandra Node))===D((Cassandra Node))
-
-      b[ecChronos Instance]---B((Cassandra Node))
-
-      B((Cassandra Node))===D((Cassandra Node))
-
-      D((Cassandra Node))---d[ecChronos Instance]
+      a[datacenter1]---A((Cassandra Node));
+      a[datacenter1]---B((Cassandra Node));
+      b[datacenter2]---C((Cassandra Node));
+      b[datacenter2]---D((Cassandra Node));
+      A((Cassandra Node))---d[ecChronos Agent];
+      B((Cassandra Node))---d[ecChronos Agent];
+      C((Cassandra Node))---d[ecChronos Agent];
+      D((Cassandra Node))---d[ecChronos Agent];
   ```
   <figcaption>Figure 1: ecChronos and Cassandra Nodes.</figcaption>
 </div>
 
 ## Concepts
 
+### Connection
+
+Once ecChronos establishes its initial connection with the `contactPoints`, it must register its control over the nodes based on the `type` property, whether JMX or CQL, to make it clear that a single instance will be responsible for managing multiple nodes. Then it would be possible to keep track of what was the last time that the EcChronos instances was able to connect with a node, also for others EcChronos instances keep track about each other's health.
+
+If type is `datacenterAware`, ecChronos will register its control over all the nodes in the specified datacenter; The `rackAware` declares that ecChronos is responsible just for a sub-set of racks in the declared list; The `hostAware` funcionality declares that ecChronos is resposible just for the specified hosts list. When connection.cql.agent.enabled is true, it must use the AgentNativeConnectionProvider class as default provider.
+
+Configuration is available on ecc.yml in the below format.
+
+```yaml
+connection:
+  cql:
+    agent:
+      type: datacenterAware
+      contactPoints:
+      - host: 127.0.0.1
+        port: 9042
+      - host: 127.0.0.2
+        port: 9042
+      datacenterAware:
+        datacenters:
+        - name: datacenter1
+        - name: datacenter2
+      rackAware:
+        racks:
+        - datacenterName: datacenter1
+          rackName: rack1
+        - datacenterName: datacenter1
+          rackName: rack2
+      hostAware:
+        hosts:
+          - host: 127.0.0.1
+            port: 9042
+          - host: 127.0.0.2
+            port: 9042
+          - host: 127.0.0.3
+            port: 9042
+          - host: 127.0.0.4
+            port: 9042
+    provider: com.ericsson.bss.cassandra.ecchronos.application.AgentNativeConnectionProvider
+```
+
+### Nodes Sync
+
+To keep track about nodes and instances, the Agent implementation uses the table nodes_sync, that declared nodes by specific ecChronos instances, so to use the Agent is mandatory to create the table below:
```

Comment:
This sentence sounds a bit weird. For instance, "that declared nodes by specific ecChronos instances" means what?

I think I understand what is to be said, but I kind of stumbles on the current wording I guess.